### DISCUSSION TITLE
Ofuscate link with NuxtLink

### DISCUSSION THREAD

**Discussion by agracia-foticos** [nuxt/nuxt#20009] [General]
We want change the behaviour of NuxtLink

We want remove href, and add

`onClick="window.location(hrefProperty)"`

the final code is

`<a href="#" onClick="window.location(hrefProperty)">Anchor text</a>`

I see that
https://nuxt.com/docs/api/components/nuxt-link#overwriting-defaults

And a custom property but i don't know how implemented it.

¿Any idea?

Thanks in advance!
Best Regards

**Comment by TheAlexLichter** [comment]
Probably the best would be writing a wrapper component around `<NuxtLink>`. But I highly advise **against that change** as you will lose SEO and a11y benefits for the links.

**Reply by agracia-foticos** [reply]
> Probably the best would be writing a wrapper component around `<NuxtLink>`. But I highly advise **against that change** as you will lose SEO and a11y benefits for the links.

We know, but in some times, we wont want  Google crawl and we need ofuscate links.

Can you give me an example please?
I don't know how implement it in best way

**Reply by TheAlexLichter** [reply]
> we wont want Google crawl a

Rather using `robots.txt` or meta tags then then 😋 

> we need ofuscate links.

For what reason exactly?


Worst case you can just do a `@click="$router.push('URL')` on a button. But as mentioned before, this is not recommended.

**Reply by agracia-foticos** [reply]
> > we wont want Google crawl a
> 
> Rather using `robots.txt` or meta tags then then yum
> 
> > we need ofuscate links.
> 
> For what reason exactly?
> 
> Worst case you can just do a `@click="$router.push('URL')` on a button. But as mentioned before, this is not recommended.

https://www.fedegomez.es/link-juice-optimizer-plugin-optimizar-crawl-budget/

For optimize the link juice :)

there isn't any option to modify actual NuxtLink to apply @click="$router.push('URL')"???

**Reply by TheAlexLichter** [reply]
> For optimize the link juice :)

Uff. I don't think that's a decent strategy. See eg. [John Mueller's tweet](https://twitter.com/JohnMu/status/1575546206238801920) or [this post](https://www.seroundtable.com/google-forget-link-juice-29859.html).

> there isn't any option to modify actual NuxtLink to apply @click="$router.push('URL')"???

No, except building a wrapper component. But that's fully up to you 👍

**Reply by agracia-foticos** [reply]
> > For optimize the link juice :)
> 
> Uff. I don't think that's a decent strategy. See eg. [John Mueller's tweet](https://twitter.com/JohnMu/status/1575546206238801920) or [this post](https://www.seroundtable.com/google-forget-link-juice-29859.html).
> 
> > there isn't any option to modify actual NuxtLink to apply @click="$router.push('URL')"???
> 
> No, except building a wrapper component. But that's fully up to you +1

Thanks for the info!!!! i see it!!!

And I am already creating my own component that simulates the button that you tell me

Have a good day!!

**Reply by AloisSeckar** [reply]
> > For optimize the link juice :)
> 
> Uff. I don't think that's a decent strategy. See eg. [John Mueller's tweet](https://twitter.com/JohnMu/status/1575546206238801920) or [this post](https://www.seroundtable.com/google-forget-link-juice-29859.html).

Not saying it is not true in the end, but where is the reason? It is just someone's statement with exactly zero arguments.

**Reply by TheAlexLichter** [reply]
@AloisSeckar In case you mean my linked posts: John Mueller is working as SEO expert at Google 😋

**Comment by priverop** [comment]
Why don't you replace NuxtLink for a div and add a @click?

Before:
```html
<NuxtLink :to="...">
  Content
</NuxtLink>
```

After:

```html
<div @click="window.location(hrefProperty)">
  Content
</div>
```

Or even with an **a** as you mentioned:

```html
<a href="#" @click="window.location(hrefProperty)">
  Content
</a>
```


If you really need to use NuxtLink create a component that wraps that component: https://ogdenstudios.xyz/blog/nuxt-link-wrapper-for-general-vue-components/
